tx-pool/src/pool.rs

```diff
@@ -465,6 +518,116 @@ impl TxPool {
         (entries, size, cycles)
     }
 
+    pub(crate) fn check_rbf(
+        &self,
+        snapshot: &Snapshot,
+        rtx: &ResolvedTransaction,
+        conflicts: &HashSet<ProposalShortId>,
+        fee: Capacity,
+        tx_size: usize,
+    ) -> Result<(), Reject> {
+        assert!(self.enable_rbf());
+        assert!(!conflicts.is_empty());
+
+        let short_id = rtx.transaction.proposal_short_id();
+        // Rule #4, new tx's fee need to higher than min_rbf_fee computed from the tx_pool configuration
+        // Rule #3, new tx's fee need to higher than conflicts, here we only check the root tx
+        if let Some(min_replace_fee) = self.calculate_min_replace_fee(conflicts, tx_size) {
+            if fee < min_replace_fee {
+                return Err(Reject::RBFRejected(format!(
+                    "Tx's current fee is {}, expect it to >= {} to replace old txs",
+                    fee, min_replace_fee,
+                )));
+            }
+        } else {
+            return Err(Reject::RBFRejected(
+                "calculate_min_replace_fee failed".to_string(),
+            ));
+        }
+
+        let pool_entries = conflicts
+            .iter()
+            .map(|id| {
+                self.get_pool_entry(id)
+                    .expect("conflict Tx should be in pool")
+            })
+            .collect::<Vec<_>>();
+
+        let mut all_statuses = pool_entries.iter().map(|e| e.status).collect::<Vec<_>>();
+
+        let conflicts = pool_entries
+            .iter()
+            .map(|e| e.inner.clone())
+            .collect::<Vec<_>>();
+
+        // Rule #2, new tx don't contain any new unconfirmed inputs
+        let mut inputs = HashSet::new();
+        for c in conflicts.iter() {
+            inputs.extend(c.transaction().input_pts_iter());
+        }
```

Comment:
no need to clone pool entries, we may use the `vec#iter` directly:

```suggestion
        // Rule #2, new tx don't contain any new unconfirmed inputs
        let mut inputs: HashSet<ckb_types::packed::OutPoint> = HashSet::new();
        for c in pool_entries.iter() {
            inputs.extend(c.inner.transaction().input_pts_iter());
        }
```

```rust
        for conflict in pool_entries.iter() {
            let descendants = self.pool_map.calc_descendants(&conflict.id);
```